app/controllers/application_controller.rb

```diff
@@ -1,10 +1,142 @@
-require './config/environment'
-
 class ApplicationController < Sinatra::Base
+set :views, Proc.new { File.join(root, "../views/") }
 
   configure do
-    set :public_folder, 'public'
-    set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "my_little_secret"
+  end
+
+    get '/' do
+
+      erb :index
+    end
+
+#------- Tweet Controls ------------#
+    get '/tweets' do
+      if Helpers.logged_in?(session)
+        @user = User.find(session[:user_id])
+        @tweets = Tweet.all
+         erb :'/tweets/tweets'
+      else
+        redirect '/users/login'
+      end
+   end
+
+   get '/tweets/new' do
+     if Helpers.logged_in?(session)
+       @user = User.find(session[:user_id])
+       erb :'/tweets/create_tweet'
+     else
+       redirect '/users/login'
+     end
+   end
+
+   get "/tweets/:id/edit" do
+     if Helpers.logged_in?(session) && Helpers.current_user(session).id == session[:user_id]
+       @tweet = Tweet.find(params[:id])
+       erb :'/tweets/edit_tweet'
+     else
+       redirect '/login'
+     end
+   end
+
+   get "/tweets/:id" do
+     if Helpers.logged_in?(session)
+       @tweet = Tweet.find(params[:id])
+       erb :'/tweets/show_tweet'
+     else
+       redirect '/login'
+     end
+   end
+
+
+   patch '/tweets/:id' do
+      @tweet = Tweet.find(params[:id])
+     if !params[:content].empty?
+       @user = @tweet.user
+       @tweet.content  = params[:content]
+       @tweet.save
+       @user.save
+       redirect "/users/#{@user.slug}"
+    else
+      redirect "/tweets/#{@tweet.id}/edit"
+    end
+   end
+
+  delete '/tweets/:id' do
+    @tweet = Tweet.find(params[:id])
+      if Helpers.current_user(session).id == @tweet.user_id
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.